src/settled-cash/CrossMarginCashMath.sol

```diff
@@ -356,45 +356,46 @@ library CrossMarginCashMath {
         //  default value of 0 is not a valid productId so the first run will always set the cache
         uint40 lastUsedProductId;
 
-        for (uint256 i; i < positions.length;) {
-            (, uint40 productId, uint64 expiry,,) = positions[i].tokenId.parseTokenId();
+        unchecked {
+            for (uint256 i; i < positions.length; ++i) {
+                (, uint40 productId, uint64 expiry,,) = positions[i].tokenId.parseTokenId();
 
-            // cache product detail if a productId differs from a previous iteration
-            if (productId != lastUsedProductId) {
-                product = _getProductDetails(grappa, productId);
+                // skip expired positions
+                if (expiry < block.timestamp) continue;
```

Comment:
hmm, we should check if the margin math allocates intrinsic value to long options that expired in ITM. If it does, then we cannot skip expired cash settled options.

Reply:
Also, for shorts this should not work because of they haven't settled the account, we dont want them to be able to use collateral that they might not have the rights to.